var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -351,6 +351,22 @@ def cmake_args(self):
             base_cmake_args.append(self.define("NC_FIND_SHARED_LIBS", False))
         return base_cmake_args
 
+    @run_after("install")
+    def patch_hdf5_pkgconfigcmake(self):
+        """
+        Incorrect hdf5 library names are put in the package config and config.cmake files
+        due to incorrectly using hdf5 target names
+        https://github.com/spack/spack/pull/42878
+        """
+
+        pkgconfig_file = join_path(self.prefix.lib64.pkgconfig, "netcdf.pc")
+        filter_file("hdf5-shared", "hdf5", pkgconfig_file)
+        filter_file("hdf5_hl-shared", "hdf5_hl", pkgconfig_file)
+
+        cmakeconfig_file = join_path(self.prefix.lib64.cmake.netCDF, "netCDFTargets.cmake")
+        filter_file("hdf5-shared", "hdf5", cmakeconfig_file)
+        filter_file("hdf5_hl-shared", "hdf5_hl", cmakeconfig_file)
```

Comment:
Thank you. Are you sure `bin/nc-config` doesn't need patching? Are you sure that the `*.pc` and `*Targets.cmake` files get installed to `lib64` on all platforms and never to `lib`? I would use [`find`](https://github.com/spack/spack/blob/983422facf5c4c7af9ef3456ff7d7f43ffa13305/lib/spack/llnl/util/filesystem.py#L1789) and [`FileFilter`](https://github.com/spack/spack/blob/983422facf5c4c7af9ef3456ff7d7f43ffa13305/lib/spack/llnl/util/filesystem.py#L439) here.

Reply:
> `nc-config`

As built for me, it doesn't report the hdf5 libraries. Indeed, I think this is a bug, but that's a separate issue
```
 ./nc-config --libs
-L/home/chm003/project-ords/spack/opt/spack/linux-rhel8-icelake/gcc-9.3.0/netcdf-c-4.9.2-6fpvnbovjwauzxdtjfdeliopq6ybu4zz/lib64 -lnetcdf
```
however, you're right that `-shared` was still lurking inside the script. `.settings` was also impacted. I've also removed the lib64 assumption.